packages/core/src/operators/use/use.operator.ts

```diff
@@ -5,7 +5,7 @@ import { HttpRequest, HttpResponse } from '../../http.interface';
 
 export const use = <I extends HttpRequest, O extends HttpRequest>
   (middleware: Middleware<I, O>, res?: HttpResponse) =>
-  (source$: Observable<I>): Observable<O> =>
+  (source$: Observable<I>) =>
     source$.pipe(
-      switchMap(req => middleware(of(req), res!, {}))
+      switchMap(req => middleware(of(req), res!, {}) as Observable<I & O>)
```

Comment:
This may not be a good way if the middleware wants to strip or override some params in the request request. For example we may have a middleware that verifies something and mutates body from:
```ts
I['body'] = { data: [1, 2, 3, 4, 5, 6, 7, 8, 9], stride: 3 };
```
to:
```ts
O['body'] = [[1, 2, 3], [4, 5, 6], [7, 8, 9]];
```
and the resulting `I & O` type may then be something like:
```ts
IO['body'] = { data: [1, 2, 3, 4, 5, 6, 7, 8, 9], stride: 3, 0: [1, 2, 3], 1: [4, 5, 6], 2: [7, 8, 9] };
```

I'd suggest leaving `as Observable<O>` here and adding a helper type for middleware creators:
```typescript
// in packages/core/src/effects/effects.interface.ts:

export interface Middleware<
  I extends HttpRequest = HttpRequest,
  O extends HttpRequest = HttpRequest,
> extends Effect<I, O> {}

/*
 * A new type explicitly defining a middleware,
 * which also accepts bare Input and Output,
 * but always returns joined `I & O` instead of `O`
 */
export interface ExtendingMiddleware<
  I extends HttpRequest = HttpRequest,
  O extends HttpRequest = HttpRequest,
> extends Middleware<I, I & O> {}
```